### PR TITLE
Rename RequestOptions to DigitalTwinsRequestOptions

### PR DESCRIPTION
**Pull request by timtay-microsoft**
We got this feedback during the Java ADT review and it is applicable over here, too

### CODE REVIEW

**Comment by azabbasi** [comment]
As long as we are talking about consistency, I think we should make the `ModelData `-> `DigitalTwinsModelData  `and `ModelProperties `-> `DigitalTwinsModelProperties `changes here as well. 
The ADT team is not going to make this change in their preview swagger but they have agreed to make the change in the GA swagger.

File: sdk/digitaltwins/Azure.DigitalTwins.Core/api/Azure.DigitalTwins.Core.netstandard2.0.cs [hunk 1]
```diff
@@ -156,9 +156,9 @@ public partial class DigitalTwinMetadata
         [System.Text.Json.Serialization.JsonExtensionDataAttribute]
         public System.Collections.Generic.IDictionary<string, object> WriteableProperties { get { throw null; } }
     }
-    public partial class ModelProperties
+    public partial class DigitalTwinsModelProperties
```
Comment (drwill-ms) [inline sdk/digitaltwins/Azure.DigitalTwins.Core/api/Azure.DigitalTwins.Core.netstandard2.0.cs:159]: I don't recall this being one that was renamed to be more specific.

**Review by drwill-ms — COMMENTED** [review]
Any changes in readme, samples/readme, change history?

File: sdk/digitaltwins/Azure.DigitalTwins.Core/api/Azure.DigitalTwins.Core.netstandard2.0.cs (hunk 1 above)
Comment (timtay-microsoft) [reply sdk/digitaltwins/Azure.DigitalTwins.Core/api/Azure.DigitalTwins.Core.netstandard2.0.cs:159]: https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/models/DigitalTwinsModelProperties.java

Looks like that is what we have in master currently

**Comment by timtay-microsoft** [comment]
> Any changes in readme, samples/readme, change history?

Samples are up to date since I did a refactor, I'll update the samples readme since they wouldn't get refactored automatically, and sure I'll add this all to the changelog

**Review by azabbasi — APPROVED** [review]

File: sdk/digitaltwins/Azure.DigitalTwins.Core/CHANGELOG.md [hunk 2]
```diff
@@ -1,7 +1,9 @@
 # Release History
 
 ## 1.0.0-preview.4 (Unreleased)
-
+-Renamed model type "ModelData" to "DigitalTwinsModelData"
```
Comment (drwill-ms) [inline sdk/digitaltwins/Azure.DigitalTwins.Core/CHANGELOG.md:4]: Empty line after headers

**Review by drwill-ms — APPROVED** [review]

**Pull request merged**